rholang/src/main/scala/coop/rchain/rholang/interpreter/match.scala

```diff
@@ -0,0 +1,273 @@
+package coop.rchain.rholang.interpreter
+
+import cats._
+import cats.data._
+import cats.implicits._
+
+import scala.annotation.tailrec
+import scala.collection.immutable.Stream
+
+object SpatialMatcher {
+  type FreeMap            = Map[Int, Par]
```

Comment:
Having worked last 2 years with `Free` monad, this will get me confused every time I will look at it :)

Reply:
Well, it's a standard use. They're free variables. Sorry for the headache.